### PR TITLE
V5 return mesh

### LINKED ISSUE #156: Return Mesh/Model when loading them
Return loaded mesh or model instead of tkaing them by reference.

### PR DESCRIPTION
**Pull request by BotellaA**
fix #156

### CODE REVIEW

File: upgrade-guide.md [hunk 1]
```diff
@@ -10,9 +10,11 @@ TODO
 
 - **Model copy**: mappings between model components have been modified to open design to arbitrary collections. It means that you can add additional information to these mappings for your own copy functions.
 
+- **Mesh loading**: mesh loading functions are now returning the mesh loaded from the filename.
+
 **How to upgrade**
```
Comment (panquez) [inline upgrade-guide.md:15]: Is there one "How to upgrade" gathering all items or one "How to upgrade for each?

File: upgrade-guide.md [hunk 2]
```diff
@@ -34,6 +36,27 @@ corner_mapping.in2out( uuid1 );
 
 - **PolygonsAroundVertex**: use PolygonVertex instead of PolygonEdge: `using PolygonsAroundVertex = absl::InlinedVector< PolygonVertex, 10 >`
 
+- **Mesh loading**: remove the mesh parameter and get the retrun value
```
[the remark this answers is not in the document]
Comment (panquez) [reply upgrade-guide.md]: remove the mesh parameter is optional. You may want it

File: include/geode/mesh/io/edged_curve_input.h [hunk 3]
```diff
@@ -38,12 +38,16 @@ namespace geode
...
+     * @param[in] type Data structure type.
      * @param[in] filename Path to the file to load.
      */
     template < index_t dimension >
-    void load_edged_curve(
-        EdgedCurve< dimension >& edged_curve, absl::string_view filename );
+    std::unique_ptr< EdgedCurve< dimension > > load_edged_curve(
+        const MeshType& type, absl::string_view filename );
+
+    template < index_t dimension >
```
Comment (panquez) [inline include/geode/mesh/io/edged_curve_input.h:48]: Add doc for the second function? For all mesh types

File: src/geode/mesh/io/edged_curve_input.cpp [hunk 4]
```diff
@@ -25,18 +25,48 @@
...
+        absl::string_view filename )
+    {
+        try
+        {
+            auto edged_curve = EdgedCurve< dimension >::create();
+            load( *edged_curve, filename );
+            return edged_curve;
         }
         catch( const OpenGeodeException& e )
         {
```
Comment (panquez) [inline src/geode/mesh/io/edged_curve_input.cpp:72]: Should that kind of code be duplicated or `load_edged_curve` without mesh type should call the one with mesh type giving default one?

Comment (BotellaA) [reply src/geode/mesh/io/edged_curve_input.cpp:72]: In this case, you have to know in this file what the default is. By having the `create()`, this is this only location where the default behavior is defined.

File: include/geode/mesh/io/edged_curve_input.h (hunk 3 above)
Comment (BotellaA) [reply include/geode/mesh/io/edged_curve_input.h:48]: Is it necessary?

Comment (panquez) [reply include/geode/mesh/io/edged_curve_input.h:48]: Ad you wish

File: src/geode/mesh/io/edged_curve_input.cpp (hunk 4 above)
Comment (panquez) [reply src/geode/mesh/io/edged_curve_input.cpp:72]: I do not understand. You can use OpenGeodeEdgedCurve< dimension >::type_name_static()?

**Pull request merged**